R/pairwise_comparisons.R

```diff
@@ -520,3 +530,206 @@ pairwise_test_bin <- function(x, group, id = NULL,
+#' @param group categorical vector which contains the group levels to compare
+#' @param id vector which contains the id information
+#' @param method a character string indicating which correlation coefficient
+#'   is to be used for the test. One of "pearson", "kendall", or "spearman",
+#'   can be abbreviated to "p", "k", or "s".
+#' @param n_distinct_value number of distinct values in `x` each `group` must
+#' have to be compared.
+#' @param digits digits to round for correlation estimate
+#' @param trailing_zeros logical indicating if trailing zeros should be included
+#' in the descriptive statistics (i.e. 0.100 instead of 0.1). Note if set to
+#' `TRUE`, output is a character vector.
+#' @param exact should exact method be used. Ignored if
+#'   `method = "pearson"` or if `method = "spearman"` and there are
+#'   ties in `x` for either `group`.
+#' @param seed seed (only used if `method = "spearman"` and there are
+#'   ties in `x` for either `group`).
+#' @param nresample a positive integer, the number of Monte Carlo replicates
+#'   used for the computation of the approximative reference distribution.
+#'   Defaults to 10000. only used if `method = "spearman"` and there are
+#'   ties in `x` for either `group`.
+#' @param verbose a logical variable indicating if warnings and messages
+#'   should be displayed.
+#' @param ... parameters passed to `stats::cor.test` or `coin:spearman_test`
+#' @return Returns a data frame with all possible pairwise correlations.
+#'
+#' @return Returns a data frame of all possible pairwise correlations
+#' with the following columns:
+#' * `Comparison` - Comparisons made
+#' * `DistinctValues` - number of distinct points
+#' * `NPairs` - number of completed pairs
+#' * `CorrEst` - correlation estimates
+#' * `CorrTest` - correlation p value
+#' * Unpasted columns if `keep_vars = TRUE`
+#' @details
+#'
+#' The p value is calculated using the [cor_test] function (see documentation
+#' for method details)
+#'
+#' If a group has less than `n_distinct_value` non-missing values that group
+#' will be excluded from the comparisons. If a specific comparison has less than
+#' `n_distinct_value` non-missing values to comparison the output will return an
+#'  estimate and the p-value set to NA.
+#'
+#' @examples
+#'
+#' data_in <- data.frame(
+#'   id = 1:10,
+#'   x = c(-2, -1, 0, 1, 2,-2, -1, 0, 1, 2),
+#'   y = c(4, 1, NA, 1, 4,-2, -1, 0, 1, 2),
+#'   z = c(1, 2, 3, 4, NA,-2, -1, 0, 1, 2),
+#'   v = c(rep(1,10)),
+#'   aa = c(1:5,NA,NA,NA,NA,NA),
+#'   bb = c(NA,NA,NA,NA,NA,1:5)
+#' )
+#' data_in_long <- tidyr::pivot_longer(data_in, -id)
+#' pairwise_test_cor(x = data_in_long$value,
+#'                   group = data_in_long$name,
+#'                   id = data_in_long$id,
+#'                   method = 'spearman')
+#'
+#'
+#' # Examples with Real World Data
+#' library(dplyr)
+#'
+#' # BAMA Assay Data Example
+#' data(exampleData_BAMA)
+#'
+#' ## Antigen Correlation
```

Comment:
Library in `dplyr` or filter won't work.
```suggestion
#' ## Antigen Correlation
#' library(dplyr)
```

Reply:
it's loaded a few lines above